sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -2499,6 +2499,39 @@ pi_result piEnqueueEventsWait(pi_queue Queue, pi_uint32 NumEventsInWaitList,
   return {};
 }
 
+pi_result piEnqueueEventsWaitWithBarrier(pi_queue Queue,
+                                         pi_uint32 NumEventsInWaitList,
+                                         const pi_event *EventWaitList,
+                                         pi_event *Event) {
+
+  assert(Queue);
+  // Get a new command list to be used on this call
+  ze_command_list_handle_t ZeCommandList = nullptr;
+  if (auto Res = Queue->Context->Device->createCommandList(&ZeCommandList))
+    return Res;
+
+  ze_event_handle_t ZeEvent = (*Event)->ZeEvent;
+
+  ze_event_handle_t *ZeEventWaitList =
```

Comment:
@smaslov-intel I implemented this consistent with the the same pointer initialization below (e.g. line 2591), but it will be memory leak on line 2518: ZE_CALL(...) - if ```zeCommandListAppendBarrier``` finishes with error code, there will be return from ```piEnqueueEventsWaitWithBarrier``` function and ```_pi_event::deleteZeEventList``` won't call.
What do you think is this a good idea to use RAII and fix this in the whole plugin as separate PR (e.g., use unique_ptr with custom deleter)?

Reply:
Good catch. Yes, I think it deserve to be fixed separately, but please add o TODO comment so it is not forgotten.

Reply:
Thanks, added.